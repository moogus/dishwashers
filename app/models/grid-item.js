import { shape, string } from 'prop-types';

export const gridItemProps = shape({
  productId: string,
  price: string,
  title: string,
  image: string,
});

export const buildModel = (gridItem = {}) => {
  const {
    productId,
    price = {},
    title,
    image,
  } = gridItem;

  const { now } = price;

  return {
    productId,
    price: `Â£${now}`,
    title,
    image,
  };
};
