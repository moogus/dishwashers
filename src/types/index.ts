export type GridItem = {
  productId: string;
  price: string;
  title: string;
  image: string;
};

export type AdditionalServices = {
  includedServices: string[];
}

export type Product = {
  title: string;
  media: Images,
  price: string;
  productInformation: string;
  displaySpecialOffer: string;
  additionalServices: AdditionalServices;
  code: string;
}

export type Price = {
  now?: string
}

export type ApiGridItem = {
  productId: string;
  price: Price;
  title: string;
  image: string;
}

export type Images = {
  altText: string;
  urls: string[]
}

export type Media = {
  images: Images;
}

export type Details = {
  productInformation: string;
}

export type ApiProduct = {
  title : string;
  media: Media;
  price?: Price;
  details?: Details;
  displaySpecialOffer : string;
  additionalServices: AdditionalServices;
  code : string;
}