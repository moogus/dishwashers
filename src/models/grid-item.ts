import { ApiGridItem, GridItem } from "../types";

export const buildModel = (apiGridItem: ApiGridItem): GridItem => {
  const {
    productId,
    price,
    title,
    image,
  } = apiGridItem;

  const { now } = price;

  return {
    productId,
    price: `Â£${now}`,
    title,
    image,
  };
};
