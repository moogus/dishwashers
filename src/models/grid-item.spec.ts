import { ApiGridItemModel } from "../types";

import { buildModel } from "./grid-item";

describe("model is built from data", () => {
  test("hydrated model is build as expected", () => {
    expect(buildModel({
      productId: "generation game",
      price: { now: "then" },
      title: "good game",
      image: "cuddly toy",
    } as ApiGridItemModel)).toMatchSnapshot();
  });
});
