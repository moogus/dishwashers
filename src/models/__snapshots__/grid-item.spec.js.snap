// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`model is built from data hydrated model is build as expected 1`] = `
Object {
  "image": "cuddly toy",
  "price": "Â£then",
  "productId": "generation game",
  "title": "good game",
}
`;
